Update Release Checklist to Clarify Best Practices When Editing Release PR

SiobhyB opened the pull request:
During a discussion that @startuptester started at p5T066-2wE-p2#comment-9434, it became clear that it can be tricky for those testing a gutenberg-mobile release to easily pinpoint changes and testing steps.

Part of the issue here is the number of PRs that need to be wrangled as part of the release, but it's also clear that the GitHub issues we link to sometimes don't contain a description of the changes and instead link to different issues themselves. Testers sometimes need to spend time clicking through multiple issues to get to the details they need.

With this PR, some extra text has been added to the release wrangling checklist to encourage release wranglers to keep testers in mind when filling out the release PR. The hope is to remind release wranglers to link directly to whichever GitHub issue actually contains the change's description/testing steps and also for them to consider adding a brief description.

### Task's text before:

> Fill in the missing parts of the gutenberg-mobile PR description

### Task's text after:

> Fill in the missing parts of the gutenberg-mobile PR description. When filling in the "Changes" section, link to the most descriptive GitHub issue for any given change and consider adding a short description. Testers rely on this section to gather more details about changes in a release.

Open to any ideas for further tweaking the text!

SiobhyB committed:
Clarify best practices when editing release PR

This commit proposes some additional text to a task surrounding editing a PR in the release wrangling checklist. The purpose is to clarify the task's purpose and ensure it's useful for the "intended audience" (testers).

fluiddot reviewed (commented):
I think it would be also useful to update the "Changes" section from the [GB-mobile PR template](https://github.com/wordpress-mobile/release-toolkit-gutenberg-mobile/blob/update/pr-edit-checklist-item/release_pull_request.md#changes), we could change the format of that section to be more descriptive, something like:
```

### Change 1
- **PR:** link-to-pr-related-to-change-1
- **Issue:** link-to-issue-related-to-change-1

short-description-describing-change-1

### Change 2
- **PR:** link-to-pr-related-to-change-2
- **Issue:** link-to-issue-related-to-change-2

short-description-describing-change-2
```

What do you think?

SiobhyB committed:
Merge branch 'develop' into update/pr-edit-checklist-item

SiobhyB committed:
Update "Changes" section in PR template

With this commit, the "Changes" section in the gutenberg-mobile PR template is updated. It now prompts the release wrangler to link to the GitHub issues related to a change, in addition to the PRs. The aim is to make the reason for each change clearer to those viewing the PR.

SiobhyB commented:
Thanks for reviewing, @fluiddot! I like the idea of improving the template and linking to the issue corresponding to a PR for clarity. I've updated the template with the `link-to-issue-related-to-change` bullet point in https://github.com/wordpress-mobile/release-toolkit-gutenberg-mobile/pull/59/commits/6b11dd8698d8da75f505c437bb080157d1267b3f.

I left out the `short-description-describing-change` item, my thinking behind that is: 

1. I'm wary of adding too much extra admin work to the release wrangler's plate, especially for releases where there are a lot of new changes. 
2. GitHub now unfurls links in lists, so we'll automatically be able to see the title for each PR and issue. Ideally, the title should contain the brief description/context that we're after.

WDYT?

fluiddot commented:
> Thanks for reviewing, @fluiddot! I like the idea of improving the template and linking to the issue corresponding to a PR for clarity. I've updated the template with the `link-to-issue-related-to-change` bullet point in [6b11dd8](https://github.com/wordpress-mobile/release-toolkit-gutenberg-mobile/commit/6b11dd8698d8da75f505c437bb080157d1267b3f).
> 
> I left out the `short-description-describing-change` item, my thinking behind that is:
> 
> 1. I'm wary of adding too much extra admin work to the release wrangler's plate, especially for releases where there are a lot of new changes.
> 2. GitHub now unfurls links in lists, so we'll automatically be able to see the title for each PR and issue. Ideally, the title should contain the brief description/context that we're after.
> 
> WDYT?

I agree that we shouldn't incur extra time on release wranglers for this but at the same time, I see a great benefit from adding a short description (I did something similar for [the `1.59.2` release](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3844)). I don't have a strong opinion so let's stick with the current changes 👍 .

fluiddot reviewed (approved):
LGTM 🎊 !
